### PR TITLE
feat(minimina): port mln amount DSL / account-ref / ISO-duration parsers

### PR DESCRIPTION
**Pull request by glyh**
## Summary

Ports the amount-DSL, account-reference, and ISO-duration parsers from the mina-local-network Python rewrite (`mln/amounts.py`) into a new pure-Rust module in the `minimina` crate: `src/app/minimina/src/amounts.rs`.

Functions (all exact integer arithmetic, no floating point for amounts):

- `amount_dsl_to_nanomina(&str) -> Result<u64, AmountParseError>` — parses `<n>mina` / `<n.frac>mina` / `<n>nanomina`; rejects >9 fractional digits as sub-nanomina precision.
- `nanomina_to_decimal_mina(u64) -> String` — formats nanomina as a 9-decimal mina string.
- `convert_balance_to_decimal_mina(&str) -> Result<String, AmountParseError>` — DSL → decimal mina.
- `parse_account_spec(&str) -> Result<(String, u32), AmountParseError>` — `tier-index` (e.g. `whale-0`) → `("whale", 0)`.
- `parse_iso_duration(&str) -> Result<f64, AmountParseError>` — `PT<number>S` → seconds.

## Notes

- **No new dependency.** The grammars are parsed by hand rather than pulling in the `regex` crate.
- Error handling uses a small module-local `AmountParseError` (implements `Display` + `std::error::Error`), mirroring the Python `INVALID_ARGUMENT` topology error.
- **No in-crate consumer yet.** Balances currently enter minimina only via a verbatim `--genesis-ledger` JSON file (copied through unparsed) or a hardcoded default balance in `genesis_ledger.rs`. Neither exposes a way to *configure* a balance via DSL. This module is the groundwork for that; it is marked `#![allow(dead_code)]` until a caller lands.

## Testing

- 28 inline unit tests mirroring the Python doctests plus edge cases (empty string, missing unit, negative-looking input, `0mina`, precision boundary, large values, two-dot inputs, wrong case).
- `rustup run 1.92.0 cargo test`: `test result: ok. 69 passed; 0 failed; 0 ignored`.
- `rustup run 1.92.0 cargo check`: clean, no warnings.

Note: this crate requires the pinned toolchain (rust 1.92 / edition2024 lock); the ambient 1.82 cannot build it.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/app/minimina/src/amounts.rs [hunk 1]
```diff
@@ -0,0 +1,441 @@
+//! # Amounts Module
+//!
+//! Pure parsing helpers ported from the mina-local-network Python rewrite
```
Comment (dkijania) [inline src/app/minimina/src/amounts.rs]: I think we don't need to mention that.

File: src/app/minimina/src/amounts.rs [hunk 2]
```diff
@@ -0,0 +1,441 @@
...
+}
+
+/// Parse a Mina amount DSL string to an integer number of nanomina.
+///
+/// Accepts `<int>mina`, a fractional `<int>.<frac>mina` (e.g. `0.25mina`), or
+/// `<int>nanomina`. Fractional mina is converted exactly via integer arithmetic;
+/// more than 9 fractional digits is sub-nanomina precision and rejected, since
+/// nanomina is the smallest representable unit.
+///
+/// Grammar: `^(\d+)(?:\.(\d+))?mina$|^(\d+)nanomina$`.
```
Comment (dkijania) [inline src/app/minimina/src/amounts.rs]: Hmm, instead or next to of that can we just use rust examples (runnable docs). I think examples would be more convenient

File: src/app/minimina/src/amounts.rs [hunk 3]
```diff
@@ -0,0 +1,441 @@
...
+    };
+    if !valid_number {
+        return Err(invalid());
+    }
+
+    inner.parse::<f64>().map_err(|_| invalid())
+}
+
+#[cfg(test)]
+mod tests {
```
Comment (dkijania) [inline src/app/minimina/src/amounts.rs:232]: I think this module is a perfect candidate to quick check tests

**Review by dkijania — APPROVED** [review]
Approved. Just nits comment

**Comment by glyh** [comment]
!ci-build-me

**Comment by amc-ie** [comment]
!ci-bypass-changelog
